src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.cpp

```diff
@@ -0,0 +1,188 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+#include "refrigerator-alarm-server.h"
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/EventLogging.h>
+#include <app/InteractionModelEngine.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/error-mapping.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::RefrigeratorAlarm;
+using namespace chip::app::Clusters::RefrigeratorAlarm::Attributes;
+using namespace chip::DeviceLayer;
+using chip::Protocols::InteractionModel::Status;
+
+RefrigeratorAlarmServer RefrigeratorAlarmServer::instance;
+
+RefrigeratorAlarmServer & RefrigeratorAlarmServer::Instance()
+{
+    return instance;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::getMaskValue(chip::EndpointId endpoint, chip::BitMask<AlarmMap> * mask)
+{
+    EmberAfStatus status = Attributes::Mask::Get(endpoint, mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: ERR: reading  mask %x", status);
+    }
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: Mask ep%d value: %" PRIu32 "", endpoint, mask->Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::getLatchValue(chip::EndpointId endpoint, chip::BitMask<AlarmMap> * latch)
+{
+    EmberAfStatus status = Attributes::Latch::Get(endpoint, latch);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: ERR: reading latch %x", status);
+    }
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: Latch ep%d value: %" PRIu32 "", endpoint, latch->Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::getStateValue(chip::EndpointId endpoint, chip::BitMask<AlarmMap> * state)
+{
+    EmberAfStatus status = Attributes::State::Get(endpoint, state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: ERR: reading state %x", status);
+    }
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: State ep%d value: %" PRIu32 "", endpoint, state->Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::setMaskValue(chip::EndpointId endpoint, chip::BitMask<AlarmMap> mask)
+{
+    chip::BitMask<AlarmMap> * currentMask = nullptr;
+    EmberAfStatus status                  = Attributes::Mask::Get(endpoint, currentMask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "Refrigerator Alarm: ERR: reading  mask %x", status);
+    }
+    if (currentMask->Raw() == mask.Raw())
```

Comment:
Why do you need to compare the raw values?   I'd expect operator== to work these objects.